spec/application-delegate-spec.js

```diff
@@ -1,14 +1,14 @@
 /** @babel */
 
-import {it, fit, ffit, fffit, beforeEach, afterEach} from './async-spec-helpers'
+import { it } from './async-spec-helpers'
```

Comment:
It's generally safer to _always_ import those. If you don't import them and you use one while trying to test something but forget to add it to the list again the test will silently "pass" and any errors will be hidden.

(The same comment applies to all the spec files where the "unused" imports are getting removed.)

Reply:
ohh I was not aware of this, thanks for rising it! I've just tried it and that behaviour is very counter intuitive 😔

@Arcanemagus do you have any context about why does the test suite behaves like this? Can we just make the test fail in these scenarios? Having to import many unused modules that could potentially be used in the test just to overcome this issue in the test runner feels wrong 😕

I'm gonna hold on merging this PR until I get some clarity about it.

The only good thing is that now that we have the linter enabled in the specs, the editor should warn about using non-declared variables 🤷‍♂️

Reply:
If you attempt to use `async` code in a version of those functions that doesn't know it needs to wait for them it will fire off the `async` code... but not wait for it. So it will race through firing all that off and reach the end seeing nothing wrong and return. Then when the `async` code returns the failures are ignored since the spec has already "finished".

It's was _very_ confusing before I discovered that. We actually wrote [jasmine-fix](https://www.npmjs.com/package/jasmine-fix) and [use it](https://github.com/AtomLinter/linter-stylelint/blob/master/spec/linter-stylelint-spec.js#L7-L10) in all of the AtomLinter packages that have `async` specs in order to avoid this issue.